### PR TITLE
Add data grid component

### PR DESCRIPTION
**Pull request by styczynski**
Add data grid based on [agGrid](https://www.ag-grid.com/)

* Custom filter
* Custom renderer for cells
* Hoverable rows

image

### CODE REVIEW

File: frontend/src/components/universal/DataGrid/CellRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    
+    constructor(props) {
+        super(props);
+        
+        this.handleMouseEnter = this.handleMouseEnter.bind(this);
+        this.handleMouseLeave = this.handleMouseLeave.bind(this);
+    }
+    
+    handleMouseEnter() {
+        if(this.props.context.enableHoverMode) {
```
Comment (Ogochi) [inline frontend/src/components/universal/DataGrid/CellRenderer.tsx]: No space after "if".

File: frontend/src/components/universal/DataGrid/CellRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            this.props.context.onRowHovered(this.props.api, this.props.rowIndex, this.props.node);
+        }
+    }
+    
+    handleMouseLeave() {
+        if(this.props.context.enableHoverMode) {
```
Comment (Ogochi) [inline frontend/src/components/universal/DataGrid/CellRenderer.tsx]: No space after "if".

File: frontend/src/components/universal/DataGrid/FilterRenderer.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        return {value: this.state.text};
+    }
+
+    setModel(model) {
+        this.state.text = model ? model.value : '';
+    }
+
+    componentMethod(message) {
+        alert(`Alert from PartialMatchFilterComponent ${message}`);
+    }
```
Comment (Ogochi) [inline frontend/src/components/universal/DataGrid/FilterRenderer.tsx]: What is it?

File: frontend/src/components/universal/DataGrid/index.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    
+    constructor(props) {
+        super(props);
+        
+        this.handleRowHovered = this.handleRowHovered.bind(this);
+        this.handleRowUnhovered = this.handleRowUnhovered.bind(this);
+    }
+    
+    handleRowUnhovered(api: GridApi, rowIndex: number, row: RowNode) {
+        if(this.props.onDataChanged) {
```
Comment (Ogochi) [inline frontend/src/components/universal/DataGrid/index.tsx]: No space after "if".

File: frontend/src/components/universal/DataGrid/index.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                        });
+                    })
+                })
+            );
+        }
+    }
+    
+    handleRowHovered(api: GridApi, rowIndex: number, row: RowNode) {
+        console.log("@handleRowHovered "+rowIndex);
+        if(this.props.onDataChanged) {
```
Comment (Ogochi) [inline frontend/src/components/universal/DataGrid/index.tsx]: No space after "if".

File: frontend/src/components/universal/DataGrid/index.tsx [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+                            hovered: false
+                        });
+                    })
+                })
+            );
+        }
+    }
+    
+    handleRowHovered(api: GridApi, rowIndex: number, row: RowNode) {
+        console.log("@handleRowHovered "+rowIndex);
```
Comment (Ogochi) [inline frontend/src/components/universal/DataGrid/index.tsx]: Do we need it?

File: frontend/src/components/universal/DataGrid/CellRenderer.tsx (hunk 1 above)
Comment (styczynski) [reply frontend/src/components/universal/DataGrid/CellRenderer.tsx]: Fixed. Future occurrences of this problem will be resolved with #56

File: frontend/src/components/universal/DataGrid/CellRenderer.tsx (hunk 2 above)
Comment (styczynski) [reply frontend/src/components/universal/DataGrid/CellRenderer.tsx]: The same as above. See #56

File: frontend/src/components/universal/DataGrid/FilterRenderer.tsx (hunk 3 above)
Comment (styczynski) [reply frontend/src/components/universal/DataGrid/FilterRenderer.tsx]: The dead code used originally for sole debugging purposes. Now it's pretty-neatly dead. Thank's for spotting that. Fixed.

File: frontend/src/components/universal/DataGrid/index.tsx (hunk 5 above)
Comment (styczynski) [reply frontend/src/components/universal/DataGrid/index.tsx]: Fixed. #56

File: frontend/src/components/universal/DataGrid/index.tsx (hunk 6 above)
Comment (styczynski) [reply frontend/src/components/universal/DataGrid/index.tsx]: `console.log` will be removed as post-debugging waste. we could setup linter to take care of the console-logs too #56

File: frontend/src/components/universal/DataGrid/index.tsx (hunk 4 above)
Comment (styczynski) [reply frontend/src/components/universal/DataGrid/index.tsx]: Fixed. #56

**Pull request merged**